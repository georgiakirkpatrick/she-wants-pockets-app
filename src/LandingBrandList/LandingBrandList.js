import React from 'react';
import Brand from '../Brand/Brand';
import ReadMoreLink from '../ReadMoreLink/ReadMoreLink';
import './LandingBrandList.css';

function LandingBrandList(props) {
    const generateBrandList = props.brands.map(brand => (
        <Brand 
            key={brand.id}
            id={brand.id}
            pathToImage={brand.pathToImage}
            imgAlt={brand.imgAlt}
            brandName={brand.brandName}
            brandLink={brand.brandLink}
        />
    ))

    return (
        <section className='LandingBrandList__section'>
            <header>
                <h3>
                    Discover new brands
                </h3>
            </header>
            <p>
                She Wants Pockets curates clothing from {props.totalBrands} from {props.totalCountries}.
                Explore our brand fact sheets and impartial supply-chain reviews to make an informed 
                decision when you vote with your money.
            </p>
            <div className='LandingBrandList__list'>
                {generateBrandList}
            </div>
            <ReadMoreLink text='All brand reviews' link='/all-categories' />
        </section>
    )
}

LandingBrandList.defaultProps = {
    brands: []
}

export default LandingBrandList;