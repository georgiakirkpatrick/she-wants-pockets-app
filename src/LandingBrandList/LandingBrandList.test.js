import React from 'react';
import ReactDOM from 'react-dom';
import LandingBrandList from './LandingBrandList';
import renderer from 'react-test-renderer';

const brands = [
    {
        id: 6860598,
        pathToImage: 'https://res.cloudinary.com/everlane/image/upload/c_scale/c_fill,dpr_1.0,f_auto,q_auto,w_auto:100:1200/v1/i/1bc014ee_3f92.jpg',
        imgAlt: `A 5'10" model is wearing a size 2 The Relaxed Air Shirt in Blue / White Stripe`,
        brandName: 'Everlane',
        brandLink: 'www.google.com'
    }
]

describe ('LandingBrandList', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<LandingBrandList />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<LandingBrandList brands={brands}/>)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})