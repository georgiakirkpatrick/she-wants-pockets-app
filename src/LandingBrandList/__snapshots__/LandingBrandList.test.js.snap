// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingBrandList renders the UI as expected 1`] = `
<section
  className="LandingBrandList__section"
>
  <header>
    <h3>
      Discover new brands
    </h3>
  </header>
  <p>
    She Wants Pockets curates clothing from 
     from 
    . Explore our brand fact sheets and impartial supply-chain reviews to make an informed decision when you vote with your money.
  </p>
  <div
    className="LandingBrandList__list"
  >
    <a
      href="www.google.com"
    >
      <div
        className="Brand"
      >
        <img
          alt="A 5'10\\" model is wearing a size 2 The Relaxed Air Shirt in Blue / White Stripe"
          src="https://res.cloudinary.com/everlane/image/upload/c_scale/c_fill,dpr_1.0,f_auto,q_auto,w_auto:100:1200/v1/i/1bc014ee_3f92.jpg"
        />
        <p>
          Everlane
        </p>
      </div>
    </a>
  </div>
  <a
    href="/all-categories"
  >
    <p
      className="ReadMoreLink"
    >
      <span>
        All brand reviews
         
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 "
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </p>
  </a>
</section>
`;
